### PR TITLE
fix revision display

### LINKED ISSUE #620: Diff viewer shows wrong versions
The diff viewer for different versions of a document seems to show not the correct differences between two selected versions.
It seems to be one version behind, so that the changes from version 0 to version 1 are shown when selecting the change from 1 to 2. That also makes it impossible to see the changes between the current and the previous version.

### PR DESCRIPTION
**Pull request by Bartzi**
fixes #620

### CODE REVIEW

File: _1327/documents/tests.py [hunk 1]
```diff
@@ -153,6 +153,50 @@ def test_revert_to_different_url(self):
 		self.assertEqual(response.status_code, 200)
 		self.assertIn(reverse('versions', args=[old_url]), response.body.decode('utf-8'))
 
+	def test_version_creation(self):
+		Document.objects.get().delete()
```
Comment (janno42) [inline _1327/documents/tests.py]: shouldn't this be `Document.objects.all().delete()`?

File: _1327/documents/tests.py [hunk 2]
```diff
@@ -153,6 +153,50 @@ def test_revert_to_different_url(self):
...
+			self.assertEqual(response.status_code, 200)
+			form = response.forms['document-form']
+
+			url_title = form.get('url_title').value
+			text_1 = 'something'
+			submit_form(text_1, form, group, sub_class)
+
+			self.assertEqual(Document.objects.count(), expected_count)
+			document = Document.objects.get(url_title=url_title)
+			response = self.app.get(reverse('minutes:edit', args=[document.url_title]), user=self.user)
```
Comment (janno42) [inline _1327/documents/tests.py]: why `minutes:edit`?

**Comment by Bartzi** [comment]
I've addressed your comments.
Regarding the server:
- we already determined that the most current version of a document is not identical to the current version, which it should be
- so we should look for all documents where the last revision is not identical to the current text and then add a new revision with the current version, that should fix it. But I recommend doing that on a test system

File: _1327/documents/utils.py [hunk 3]
```diff
@@ -49,12 +49,13 @@ def handle_edit(request, document, formset=None, initial=None, creation_group=No
...
 
 			# save the document and also save the user and the comment the user added
-			with transaction.atomic(), revisions.create_revision():
+			with transaction.atomic():
 				document.handle_edit(cleaned_data)
-				document.save()
-				document.save_formset(formset)
-				revisions.set_user(request.user)
-				revisions.set_comment(cleaned_data['comment'])
+				with revisions.create_revision():
```
Comment (janno42) [inline _1327/documents/utils.py]: can you please add a comment what this does and why `revisions.create_revision()` has to be where it is?

Comment (janno42) [reply _1327/documents/utils.py]: meaning why it needed to be moved from the line above

**Pull request merged**